pkg/controller/node/nodecontroller_test.go

```diff
@@ -1265,37 +1267,43 @@ func TestMonitorNodeStatusEvictPodsWithDisruption(t *testing.T) {
 		if err := nodeController.monitorNodeStatus(); err != nil {
 			t.Errorf("%v: unexpected error: %v", item.description, err)
 		}
-		// Give some time for rate-limiter to reload
-		time.Sleep(500 * time.Millisecond)
-
 		for zone, state := range item.expectedFollowingStates {
 			if state != nodeController.zoneStates[zone] {
 				t.Errorf("%v: Unexpected zone state: %v: %v instead %v", item.description, zone, nodeController.zoneStates[zone], state)
 			}
 		}
-		zones := testutil.GetZones(fakeNodeHandler)
-		for _, zone := range zones {
-			// Time for rate-limiter reloading per node.
-			time.Sleep(50 * time.Millisecond)
-			nodeController.zonePodEvictor[zone].Try(func(value TimedValue) (bool, time.Duration) {
+		podEvicted = false
+		// Infinite loop, used for retrying in case ratelimiter fails to reload for Try function.
+		// this breaks when we have the status that we need for test case.
+		// TODO: Add some timeout for infinite loop to break.
+		for {
```

Comment:
> Except this for-loop being infinite smells wrong. We can certainly retry few times.

@gmarek - So, I was having a problem in coming up with number. It worked fine with 50 being max number of retries on my system but when I tried on another system, it failed. As you suggested, whatever the number we choose, it may not be perfect based on CPU load. So, if you are ok with timeout as a limit, the timeout being 5 min, I can make that change.